c++/src/kj/io.h

```diff
@@ -55,10 +55,10 @@ class InputStream {
   // If the InputStream can't produce minBytes, it MUST throw an exception, as the caller is not
   // expected to understand how to deal with partial reads.
 
-  virtual size_t tryRead(void* buffer, size_t minBytes, size_t maxBytes) = 0;
+  virtual size_t tryRead(ArrayPtr<byte> buffer, size_t minBytes) = 0;
   // Like read(), but may return fewer than minBytes on EOF.
 
-  inline void read(void* buffer, size_t bytes) { read(buffer, bytes, bytes); }
+  inline void read(ArrayPtr<byte> buffer) { read(buffer, buffer.size()); }
```

Comment:
Since this is just falling back to `read(buffer, minBytes)` I'm really not sure it's all that useful? We could just specify a default value for `minBytes` that is set to max(size_t)` with the interpretation that any `minBytes` larger than `buffer` can be handled as `minBytes = buffer.size()`

Reply:
I think that would impose additional complexity on all implementations just to eliminate this one inline method. Doesn't seem worth it.